src/Workspaces/Core/Portable/Workspace/Solution/SolutionState.cs

```diff
@@ -464,16 +464,18 @@ public SolutionState RemoveProjects(ArrayBuilder<ProjectId> projectIds)
             dependencyGraph: newDependencyGraph);
     }
 
-    private static void AddLanguageCount(ref TemporaryArray<(string language, int count)> languageCounts, string language, int amount)
+    private static void AddLanguageCountDelta(ref TemporaryArray<(string language, int count)> languageCountDeltas, string language, int amount)
     {
-        var index = languageCounts.IndexOf(static (c, language) => c.language == language, language);
+        Contract.ThrowIfFalse(amount is -1 or +1);
+
+        var index = languageCountDeltas.IndexOf(static (c, language) => c.language == language, language);
         if (index < 0)
         {
-            languageCounts.Add((language, amount));
+            languageCountDeltas.Add((language, amount));
```

Comment:
note: this is still an issue for me namely that if we do *not* find the item we shouldn't be setting this to -1.  THere should be a contract call that it's only ok to not find the item in the +1 ccase.

Reply:
The array starts empty. When we are removing project we will not fine the item in it and we do want to set the value -1.

Reply:
got it.  thanks.